### ISSUE TITLE
Add configuration option to turn off bail

### ISSUE THREAD

**Issue by JSmits-utwente** [stryker-mutator/stryker-js#2996]
**Is your feature request related to a problem? Please describe.**
For my graduation assignment I would like to use StrykerJS's JSON reporter to export mutation information that I need. In particular I need to know for each mutant which test cases have _coverage_ over that mutant, and which test cases _kill_ that mutant.
Currently StrykerJS uses a feature called _bail_, where mutant execution stops once one test case has managed to kill that mutant. This increases performance, but also means the JSON report only ever lists a maximum of one test case as the killer of a mutant, even if there are more.

**Describe the solution you'd like**
Add an option to StrykerJS's configuration to allow toggling bail. The default for that option can then still be to turn it on. Something like:
`enableBail` [`boolean`]
Default: `true`
Command line: `-- enableBail true`
Config file: `"enableBail": true`

**Describe alternatives you've considered**
N/A

**Comment by Edgpaez** [comment]
This seems to be supported already (although I haven't tested myself):
1. jest-runner seems to have an [enableBail option](https://github.com/stryker-mutator/stryker-js/blob/543fa2df8eeb88b753ac1d712fa857e0afb5f5d5/packages/jest-runner/schema/jest-runner-options.json#L30)
2. You could pass options to jest using  [`testRunnerNodeArgs`](https://stryker-mutator.io/docs/stryker-js/configuration#testrunnernodeargs-string) or using [config](https://stryker-mutator.io/docs/stryker-js/jest-runner#jestconfig-object) directly

**Comment by nicojs** [comment]
Thanks for pitching in @Edgpaez ❤

You are correct, but the problem is that all test runner plugins a.t.m. only report back the first killed test. As you can see in the `MutantRunResult` interface:

https://github.com/stryker-mutator/stryker-js/blob/5466cfb4fce3de6c7b7817c2c37812215848000a/packages/api/src/test-runner/mutant-run-result.ts#L14-L28

In order to implement this feature, we should:
1. Change the `MutantRunResult` interface:
  ```diff
  export interface KilledMutantRunResult {
    status: MutantRunStatus.Killed;
  - killedBy: string;
  + killedBy: string[];
    failureMessage: string;
    nrOfTests: number;
  }
  ```
2. Update all `TestRunner` plugins to support this new interface.
3. Update the core to support this new interface
    * We should keep it backward compatible (still support plain strings), otherwise it would be a breaking change
    * Don't forget to update the json reporting https://github.com/stryker-mutator/stryker-js/blob/5466cfb4fce3de6c7b7817c2c37812215848000a/packages/core/src/reporters/mutation-test-report-helper.ts#L69
4. Add new config option `--bail` (name pending)
5. Add support for `--bail` in each test runner. This is pretty straightforward, as jasmine, cucumber, mocha, and jest are supporting it already.
6. (optional): remove support for `jest.enableBail`, as it is replaced by `--bail`.

All in all: not difficult to do, but it will take some time to implement 🤷‍♂️

**Comment by rouke-broersma** [comment]
Fyi we are naming this option `--disableBail` because it is enabled by default and the only reason to specify it is to disable it.

**Comment by james-perretta** [comment]
Are you looking for help with this issue? I was planning on patching my local copy of stryker-js anyway so that I can get the data I need.

**Comment by nicojs** [comment]
The last I heard is that @Edgpaez is working on it. He is active in our slack, so maybe you can contact him there? Thanks for any help you can give.

**Comment by nicojs** [comment]
>     1. Change the `MutantRunResult` interface:
> 
> 
> ```diff
> export interface KilledMutantRunResult {
>   status: MutantRunStatus.Killed;
> - killedBy: string;
> + killedBy: string[];
>   failureMessage: string;
>   nrOfTests: number;
> }
> ```
> 
>     1. Update all `TestRunner` plugins to support this new interface.
> 
>     2. Update the core to support this new interface
>        
>        * We should keep it backward compatible (still support plain strings), otherwise it would be a breaking change
>        * Don't forget to update the json reporting https://github.com/stryker-mutator/stryker-js/blob/5466cfb4fce3de6c7b7817c2c37812215848000a/packages/core/src/reporters/mutation-test-report-helper.ts#L69

**Comment by nicojs** [comment]
Support for `--disableBail` was merged in #3074 and released in StrykerJS v5.4
